src/components/MapController/MapController.js

```diff
@@ -19,51 +38,98 @@ var MapController = React.createClass({
+  //parameters like logscale or palette.
   updateSelection: function (param, selection) {
     var update = {}; update[param] = selection;
     this.setState(update);
   },
 
+  //this function updates the timestamp displayed on the map and triggers 
+  //a rerender.
+  //timeidx is a stringified object with a resolution
+  //(monthly, annual, seasonal) and an index (0-11). For example
+  //{timeres: monthly, timeidx: 3} would represent April. 
+  //It's stringified because Selector won't pass an object.
   updateTime: function (timeidx) {
     this.setState({
       timeidx: timeidx,
       wmstime: this.state.times[timeidx],
     });
   },
 
-  updateDataset: function (uniqueId) {
-    // Updates dataset in state. Updates time value to match new dataset
-
-    this.selectedDataset = this.props.meta.filter(function (el) {
-      return el.unique_id === uniqueId;
-    })[0];
-
-    this.requestTimeMetadata(uniqueId).then(response => {
-      this.selectedDataset.times = response.data[uniqueId].times;
+  //this function stores a dataset selected by the user to state, 
+  //triggering a rerender to show the new dataset.
+  //a "dataset" in this case is not a specific file. It is a
+  //variable + emissions + model + period + run combination. Timestamps for
+  //a "dataset" may be spread across up to three files (one annual, one 
+  //seasonal, one monthly). MapController stores the parameters of the dataset
+  //in state, but doesn't select (or store in state) a specific file with a 
+  //specific unique_id until rendering, when it needs to pass an exact file
+  //and timestamp to the viewer component CanadaMap.
+  updateDataset: function (dataset) {
+    dataset = JSON.parse(dataset);
+    var run = dataset.ensemble_member;
+    var start_date = dataset.start_date;
+    var end_date = dataset.end_date;
+    
+    //generate the list of available times for this variable+run+period
+    //which may include multiple files of different time resolutions 
+    //(annual, seasonal, or monthly).
+    var times = {};
+    var timesPromises = [];
+        
+    var datasets = _.filter(this.props.meta, 
+        {"ensemble_member": run, "start_date": start_date, "end_date": end_date });
+    
+    for(var i = 0; i < datasets.length; i++) {
+      timesPromises.push(this.requestTimeMetadata(datasets[i].unique_id));
+    }
+   
+    Promise.all(timesPromises).then(responses => {
+      for(var i = 0; i < responses.length; i++) {
+        var id = Object.keys(responses[i].data)[0];
+        for(var timeidx in responses[i].data[id].times) {
+          var idxString = JSON.stringify({
+            "timescale": responses[i].data[id].timescale,
+            "timeidx": timeidx,
+          });
+          times[idxString] = responses[i].data[id].times[timeidx];
+        }
+      }
+      //select an arbitrary index to display initially. It will be a
+      //0th index, but could be January, winter, or Annual.
+      var startingIndex = Object.keys(times)[0];
```

Comment:
Danger! A hash is unordered. Hash keys are not guaranteed to be returned in the order they are stored. So the value initially displayed could be any of the values stored. 
If that's OK behaviour, this code is OK. If not (and it seems a bit weird to allow an effectively random initial value), you need to add some kind of order-awareness -- perhaps the simplest would be to store the first index as it is generated.